File: src/main/java/com/sfh/pokeRogueBot/model/decisions/LearnMoveDecision.java
package com.sfh.pokeRogueBot.model.decisions;

import com.sfh.pokeRogueBot.model.enums.LearnMoveReasonType;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class LearnMoveDecision {

    private boolean newMoveBetter;
    private int moveIndexToReplace;
    private LearnMoveReasonType reason;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/decisions/LearnMoveDecision.kt
package com.sfh.pokeRogueBot.model.decisions

import com.sfh.pokeRogueBot.model.enums.LearnMoveReasonType

data class LearnMoveDecision(
    val isNewMoveBetter: Boolean,
    val moveIndexToReplace: Int,
    val reason: LearnMoveReasonType
)